tests/Guesser/FilterTypeGuesserTest.php

```diff
@@ -35,6 +35,9 @@ public function testThrowsOnMissingField()
 
         $class = 'My\Model';
         $property = 'whatever';
+
+        $this->metadata->hasAssociation($property)->willReturn(false);
+
```

Comment:
Why did you add this line exactly?

Reply:
method `metadata->hasAssociation` must return a boolean _(typed since https://github.com/doctrine/mongodb-odm/pull/1846 I think)_ Without this line, test fail.


Test log https://travis-ci.org/sonata-project/SonataDoctrineMongoDBAdminBundle/jobs/593439895#L498

<details>
1) Sonata\DoctrineMongoDBAdminBundle\Tests\Guesser\FilterTypeGuesserTest::testThrowsOnMissingField
Failed asserting that exception of type "TypeError" matches expected exception "Sonata\DoctrineMongoDBAdminBundle\Model\MissingPropertyMetadataException". Message was: "Return value of Double\Doctrine\ODM\MongoDB\Mapping\ClassMetadata\P8::hasAssociation() must be of the type bool, null returned" at
/home/travis/build/sonata-project/SonataDoctrineMongoDBAdminBundle/src/Guesser/FilterTypeGuesser.php:48
/home/travis/build/sonata-project/SonataDoctrineMongoDBAdminBundle/tests/Guesser/FilterTypeGuesserTest.php:43
</details>

cf:
- https://github.com/romainnorberg/SonataDoctrineMongoDBAdminBundle/blob/master/src/Guesser/FilterTypeGuesser.php#L48

Reply:
Oh ok, there was probably no type declaration before in `ClassMetadata`, so it was fine with returning `null`. I think your fix is ok 👍